gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriter.java

```diff
@@ -231,7 +231,13 @@ public boolean apply(FileStatus input) {
         if (isInstrumentationEnabled()) {
           copier.withCopySpeedMeter(this.copySpeedMeter);
         }
-        this.bytesWritten.addAndGet(copier.copy());
+        long numBytes = copier.copy();
+        long fileSize = copyableFile.getFileStatus().getLen();
```

Comment:
I think this check will break if encryption is turned on in the OutputStream -- can you make sure there is a test verifying this?

Reply:
You're right, changed the check configurable and defaulting to false.